lib/mixlib/shellout/windows.rb

```diff
@@ -313,6 +320,38 @@ def self.find_executable(path)
         def self.executable?(path)
           File.executable?(path) && !File.directory?(path)
         end
+
+        def self.kill_process_tree(pid, wmi, logger)
+          wmi.query("select * from Win32_Process where ParentProcessID=#{pid}").each do |instance|
```

Comment:
I thought the thing that motivated this fix was wmi was freezing up. Or was that something else?